### ISSUE TITLE
Creating a second /augeas/context node makes augeas unusable

### ISSUE THREAD

**Issue by lutter** [hercules-team/augeas#374]
In augtool:
```
augtool> set /augeas/context[2] /augeas
augtool> ls
error: Failed to execute command
Not enough arguments for ls
augtool> print /augeas
error: Too many matches for path expression
There are 2 nodes matching /augeas/context, expecting one
augtool> match /augeas
error: Too many matches for path expression
There are 2 nodes matching /augeas/context, expecting one
augtool> rm /augeas/context
error: Too many matches for path expression
There are 2 nodes matching /augeas/context, expecting one
```

We should change the internal search expression that we use to look up `/augeas/context` to `/augeas/context[last()]` .. that would also provide an easy way to maintain a stack of contexts where push would be `set /augeas/context[last()+1] foo` and pop would be `rm /augeas/context[last()]`

**Comment by lutter** [comment]
There's a similar problem if `/augeas/context` gets set to something invalid like
```
augtool> set /augeas/context 'nonsense['
augtool> match /files
error: Invalid path expression
error: illegal string literal
nonsense[|=|
augtool> rm /augeas/context
error: Invalid path expression
error: illegal string literal
nonsense[|=|
augtool> set /augeas/context /files
augtool> match /files
/files = (none)
```

The last `set` succeeds because the context path gets special cased so that users can fix a screwed up context. It would probably be better to refactor the code in a way that anything involving absolute path expressions would work regardless of whether the context is set correctly or not since the context has no influence on how absolute path expressions get evaluated.

In a nutshell, `tree_root_ctx` should be moved into `pathx.c` and only called when absolutely necessary.
